addons/mail/static/src/models/message_view.js

```diff
@@ -21,24 +21,45 @@ registerModel({
         /**
          * @param {MouseEvent} ev
          */
-        onClick(ev) {
+        async onClick(ev) {
             if (ev.target.closest('.o_channel_redirect')) {
-                this.messaging.openProfile({
+                // avoid following dummy href
+                ev.preventDefault();
+                let channel = this.messaging.models['Thread'].findFromIdentifyingData({
                     id: Number(ev.target.dataset.oeId),
                     model: 'mail.channel',
                 });
-                // avoid following dummy href
+                if(!channel) {
+                    channel = this.messaging.models['Thread'].insert({
+                        id: Number(ev.target.dataset.oeId),
+                        model: 'mail.channel',
+                    });
+                }
+                if(!channel.isJoined) {
+                    await channel.join();
+                    channel.update({
+                        isServerPinned: true,
+                        isJoined: true,
+                    });
+                }
+                channel.open({ focus: true });
+                return;
+            }
+            else if (ev.target.closest('.o_mail_redirect')) {
                 ev.preventDefault();
+                this.messaging.openChat({
+                    partnerId: Number(ev.target.dataset.oeId)
+                });
                 return;
             }
             if (ev.target.tagName === 'A') {
+                // avoid following dummy href
+                ev.preventDefault();
```

Comment:
why this move? When clicking on a normal link not detected as an internal action, we shouldn't prevent the defautl action